python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -73,9 +73,15 @@ def __init__(
         asset_graph: BaseAssetGraph,
         evaluation_time: Optional[datetime] = None,
         logger: Optional[logging.Logger] = None,
+        loading_context: Optional[LoadingContext] = None,
     ):
         self._instance = instance
         self._loaders = {}
+
+        # only use self as a loading context if one is not passed in
+        # in the future, we can make CachingInstanceQueryer not be a LoadingContext
+        self._loading_context = loading_context or self
```

Comment:
what about this bit ? can raise it on the PR that makes this a loading context but I have similar concerns about the subtle behavior differences this could cause